mono/mini/mini-trampolines.c

```diff
@@ -792,6 +792,7 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 		 */
 		if (domain_jit_info (domain)->jump_target_got_slot_hash) {
 			GSList *list, *tmp;
+			m = mono_method_to_shared (m);
```

Comment:
Would be better to move this whole block of code to mini_patch_jump_sites ().

Reply:
Also..mini_patch_jump_sites locks more briefly, but leaks the list?
This will also make one instead of two conversion to shared.
Merge them with another PR or in this PR?

Reply:
Looks like my "bikeshed" broke things. I'll undo that.